### ISSUE TITLE
Add `gsl-lite` library

### ISSUE THREAD

**Issue by rynkowsg** [ruslo/hunter#571]
GitHub: https://github.com/martinmoene/gsl-lite

C++ Core Guidelines: https://github.com/isocpp/CppCoreGuidelines
Stroustrup's talk about GSL: https://www.youtube.com/watch?v=1OEu9C51K2A

gsl-lite is a single-file header-only variant of Microsoft's implementation of the Guideline Support Library (GSL) adapted for C++98, C++03. It should also work when compiled as C++11, C++14.

The Guideline Support Library (GSL) contains functions and types that are suggested for use by the C++ Core Guidelines maintained by the Standard C++ Foundation. The library includes types like owner<>, not_null<>, span<>, string_span and others.

**Comment by xsacha** [comment]
Since this is a single header file (gsl_lite.h, I use it in my project), does it even make sense to have a package for it?
I know the same is done for the amalgamated sqlite.h but I just end up adding the file to my project anyway.

**Comment by ruslo** [comment]
> does it even make sense to have a package for it?

Yes it make sense because of unified integration interface for all package (source, header-only, one file, two file, etc.): `find_package(Foo CONFIG REQUIRED)`

@xsacha What is the alternative and how do you use it now?

**Comment by xsacha** [comment]
I just stick the header in my include folder and call it a day :)

**Comment by ruslo** [comment]
If your project is a library you may have problems with distribution. For example if somebody will install GSL in a "normal" way and then you will install your library with copy of GSL - you will overwrite the "good" version. Another example is when somebody do the same trick and you will have two versions of GSL in include paths. As a general note any kind of "local" based approaches is broken by design (git submodules, add_subdirectory(third_party/Foo), etc.) and can be used only as a temporary solutions for development purposes, not for releases. Hunter do use "global" approach when everything installed into some root directory and all projects using that directory. This fits perfectly for how CMake's find_package works: any number of projects/different places in code which call find_package will find the same "global" version.

**Comment by xsacha** [comment]
Fair enough, it will work well integrating with other hunter projects.
